src/image/PhotoCalib.cc

```diff
@@ -285,6 +289,18 @@ class PhotoCalibFactory : public table::io::PersistableFactory {
     read(InputArchive const &archive, CatalogVector const &catalogs) const override {
         table::BaseRecord const &record = catalogs.front().front();
         PhotoCalibSchema const &keys = PhotoCalibSchema::get();
+        int version = -1;
+        try {
+            std::string versionName = "version";
+            auto versionKey = record.getSchema().find<int>(versionName);
+            version = record.get(versionKey.key);
+        } catch (const pex::exceptions::NotFoundError &) {
+            // un-versioned files are version 0
+            version = 0;
+        }
+        if (version != 0) {
+            throw(pex::exceptions::TypeError("Unsupported version: " + std::to_string(version)));
+        }
```

Comment:
This block of new code seems like a good candidate for a separate method; even if it's only used once, just having a call like `int version = getVersion(record);` in `read` will improve code clarity a bit.